common/persistence/dataInterfaces.go

```diff
@@ -1073,6 +1073,8 @@ type (
 		MinEventID int64
 		// Get the history nodes upto MaxEventID.  Exclusive.
 		MaxEventID int64
+		// If set, identifies event ID at which reset has to be performed.
+		ResetFromID int64
```

Comment:
I don't think we should introduce this concept of ResetFromID as part of ReadHistoryBranchRequest.  This API is very generic and used from a lot of different places, and introducing ResetFromID as the parameter just makes it harder to use.
The decision to ignore events after reset point if history is corrupted is very specific to reset use case and should live in the reset code path.

Reply:
I agree that it's a bit too specific, let's discuss alternatives offline.